ev3dev/core.py

```diff
@@ -955,6 +958,94 @@ def wait_while(self, s, timeout=None):
         """
         return self.wait(lambda state: s not in state, timeout)
 
+    def set_power(self, power):
+        self.speed_sp = power
+
+    def set_position_rotations(self, power, rotations):
+        if power > 0:
+            self.position_sp = self.position + int(rotations * self.count_per_rot)
+        else:
+            self.position_sp = self.position - int(rotations * self.count_per_rot)
+
+    def set_position_degrees(self, power, degrees):
+        rotations = float(degrees / self.count_per_rot)
+        self.set_position_rotations(power, rotations)
+
+    def set_brake(self, brake):
+        if brake:
+            self.stop_action = self.STOP_ACTION_BRAKE
+        else:
+            self.stop_action = self.STOP_ACTION_COAST
+
+    def set_time(self, ms):
+        self.time_sp = ms
+
+    def on_for_rotations(self, power, rotations, brake):
+        """
+        Rotate the motor at 'power' for 'rotations'
+        """
+        self.set_power(power)
+        self.set_position_rotations(power, rotations)
+        self.set_brake(brake)
+        self.run_to_abs_pos()
+
+    def on_for_degrees(self, power, degrees, brake):
+        """
+        Rotate the motor at 'power' for 'degrees'
+        """
+        self.set_power(power)
+        self.set_position_degrees(power, degrees)
+        self.set_brake(brake)
+        self.run_to_abs_pos()
+
+    def on_for_seconds(self, power, seconds, brake):
+        """
+        Rotate the motor at 'power' for 'seconds'
+        """
+        self.set_power(power)
+        self.set_time(seconds * 1000)
+        self.set_brake(brake)
+        self.run_timed()
+
+    def on(self, power):
+        """
+        Rotate the motor at 'power' for forever
+        """
+        self.set_power(power)
+        self.run_forever()
+
+    def off(self, brake):
+        self.set_brake(brake)
+        self.stop()
+
+    def get_rotations(self):
```

Comment:
How about using python properties for these?

Reply:
I hate decorators :)  Since we do use `@property` elsewhere though I guess it does make sense to use them here....will fix

Reply:
```python
def rotations(self):
    pass

rotations = property(rotations)
```

No decorator. :laughing: